pkg/encoding/fix.go

```diff
@@ -95,3 +100,13 @@ func fixConfigMaps(app *spec.App) error {
 	}
 	return nil
 }
+
+func fixJobs(app *spec.App) error {
+	for index, job := range app.Jobs {
+		// populate job name if not present
+		if job.Name == "" {
+			app.Jobs[index].Name = "job-" + strconv.Itoa(index+1)
```

Comment:
Don't add 1 to index, even Kubernetes starts the indexing from 0, e.g. in case of StatefulSets (TIL) :P